### PR TITLE
GEOMESA-2017 Bumping versions of core dependencies for 1.4

### PR DESCRIPTION
**Pull request by elahrvivaz**

### CODE REVIEW

File: build/cqs.tsv [hunk 1]
```diff
@@ -4,9 +4,8 @@ com.beust:jcommander	1.48	compile
 com.carrotsearch:hppc	0.7.2	compile
 com.clearspring.analytics:stream	2.9.2	compile
 com.drewnoakes:metadata-extractor	2.8.1	compile
-com.esotericsoftware.kryo:kryo	2.21	compile
+com.esotericsoftware.kryo:kryo	2.24.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: i wonder if we're going to have any problems with kryo... was there a reason to bump the version?
https://rawgit.com/EsotericSoftware/kryo/master/compat_reports/kryo/2.22_to_2.23.0/compat_report.html
https://rawgit.com/EsotericSoftware/kryo/master/compat_reports/kryo/2.23.0_to_2.24.0/compat_report.html

File: build/cqs.tsv [hunk 2]
```diff
@@ -75,6 +77,12 @@ io.netty:netty-codec	4.0.41.Final	compile
 io.netty:netty-common	4.0.41.Final	compile
 io.netty:netty-handler	4.0.41.Final	compile
 io.netty:netty-transport	4.0.41.Final	compile
+it.geosolutions.imageio-ext:imageio-ext-gdal-bindings	1.9.2	compile
+it.geosolutions.imageio-ext:imageio-ext-gdalframework	1.1.15	compile
+it.geosolutions.imageio-ext:imageio-ext-gdalvrt	1.1.17	compile
+it.geosolutions.imageio-ext:imageio-ext-geocore	1.1.15	compile
+it.geosolutions.imageio-ext:imageio-ext-streams	1.1.15	compile
+it.geosolutions.imageio-ext:imageio-ext-utilities	1.1.15	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: we probably need to exclude these right?

File: build/cqs.tsv (hunk 1 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: it looks like spark has bumped to twitter-chill-0.8, which includes kryo 3.0.3 - we've still got twitter-chill pegged to 0.5, should probably update that

File: build/cqs.tsv [hunk 3]
```diff
@@ -4,9 +4,8 @@ com.beust:jcommander	1.48	compile
 com.carrotsearch:hppc	0.7.2	compile
 com.clearspring.analytics:stream	2.9.2	compile
 com.drewnoakes:metadata-extractor	2.8.1	compile
-com.esotericsoftware.kryo:kryo	2.21	compile
-com.esotericsoftware.minlog:minlog	1.2	compile
-com.esotericsoftware.reflectasm:reflectasm	shaded:1.07	compile
+com.esotericsoftware:kryo-shaded	3.0.3	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14202

File: build/cqs.tsv [hunk 4]
```diff
@@ -4,9 +4,8 @@ com.beust:jcommander	1.48	compile
...
+com.esotericsoftware:minlog	1.3.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14203

File: build/cqs.tsv [hunk 5]
```diff
@@ -34,7 +33,7 @@ commons-collections:commons-collections	3.2.2	compile
 commons-configuration:commons-configuration	1.6	compile
 commons-digester:commons-digester	1.8	compile
 commons-httpclient:commons-httpclient	3.1	compile
-commons-io:commons-io	2.4	compile
+commons-io:commons-io	2.5	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14204

File: build/cqs.tsv [hunk 6]
```diff
@@ -47,7 +46,7 @@ com.sun.xml.bind:jaxb-impl	2.2.11	compile
 com.thoughtworks.paranamer:paranamer	2.6	compile
 com.typesafe:config	1.2.1	compile
 com.typesafe.scala-logging:scala-logging_2.11	3.1.0	compile
-com.vividsolutions:jts	1.13	compile
+com.vividsolutions:jts-core	1.14.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14205

File: build/cqs.tsv [hunk 7]
```diff
@@ -75,36 +74,36 @@ io.netty:netty-codec	4.0.41.Final	compile
...
+it.geosolutions.jaiext.piecewise:jt-piecewise	1.0.15	compile
+it.geosolutions.jaiext.rescale:jt-rescale	1.0.15	compile
+it.geosolutions.jaiext.rlookup:jt-rlookup	1.0.15	compile
+it.geosolutions.jaiext.scale:jt-scale	1.0.15	compile
+it.geosolutions.jaiext.stats:jt-stats	1.0.15	compile
+it.geosolutions.jaiext.translate:jt-translate	1.0.15	compile
+it.geosolutions.jaiext.utilities:jt-utilities	1.0.15	compile
+it.geosolutions.jaiext.vectorbin:jt-vectorbin	1.0.15	compile
+it.geosolutions.jaiext.warp:jt-warp	1.0.15	compile
+it.geosolutions.jaiext.zonal:jt-zonal	1.0.15	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14206

File: build/cqs.tsv [hunk 8]
```diff
@@ -115,20 +114,20 @@ net.minidev:json-smart	2.2.1	compile
 net.sf.geographiclib:GeographicLib-Java	1.44	compile
 net.sf.trove4j:trove4j	3.0.3	compile
 org.antlr:antlr4-runtime	4.5	compile
-org.apache.accumulo:accumulo-core	1.7.2	compile
-org.apache.accumulo:accumulo-fate	1.7.2	compile
-org.apache.accumulo:accumulo-start	1.7.2	compile
+org.apache.accumulo:accumulo-core	1.8.1	compile
+org.apache.accumulo:accumulo-fate	1.8.1	compile
+org.apache.accumulo:accumulo-start	1.8.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14208

File: build/cqs.tsv [hunk 9]
```diff
@@ -115,20 +114,20 @@ net.minidev:json-smart	2.2.1	compile
...
 org.apache.arrow:arrow-format	0.6.0	compile
 org.apache.arrow:arrow-memory	0.6.0	compile
 org.apache.arrow:arrow-vector	0.6.0	compile
 org.apache.avro:avro	1.7.5	compile
 org.apache.camel:camel-core	2.15.1	compile
 org.apache.camel:camel-scala	2.15.1	compile
 org.apache.commons:commons-compress	1.4.1	compile
-org.apache.commons:commons-csv	1.0	compile
+org.apache.commons:commons-csv	1.5	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14209

File: build/cqs.tsv [hunk 10]
```diff
@@ -137,10 +135,10 @@ org.apache.curator:curator-framework	2.4.0	compile
...
 org.apache.curator:curator-recipes	2.4.0	compile
 org.apache.curator:curator-recipes	2.7.1	compile
-org.apache.hbase:hbase-client	1.2.3	compile
-org.apache.hbase:hbase-common	1.2.3	compile
-org.apache.hbase:hbase-protocol	1.2.3	compile
-org.apache.hbase:hbase-server	1.2.3	compile
+org.apache.hbase:hbase-client	1.3.1	compile
+org.apache.hbase:hbase-common	1.3.1	compile
+org.apache.hbase:hbase-protocol	1.3.1	compile
+org.apache.hbase:hbase-server	1.3.1	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14210

File: build/cqs.tsv [hunk 11]
```diff
@@ -150,49 +148,50 @@ org.apache.parquet:parquet-encoding	1.9.0	compile
 org.apache.parquet:parquet-format	2.3.1	compile
 org.apache.parquet:parquet-hadoop	1.9.0	compile
 org.apache.parquet:parquet-jackson	1.9.0	compile
-org.apache.thrift:libthrift	0.9.1	compile
+org.apache.thrift:libthrift	0.9.3	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14211

File: build/cqs.tsv [hunk 12]
```diff
@@ -150,49 +148,50 @@ org.apache.parquet:parquet-encoding	1.9.0	compile
...
 org.apache.xml:xml-commons-resolver	1.2	compile
-org.apache.zookeeper:zookeeper	3.4.5	compile
+org.apache.zookeeper:zookeeper	3.4.10	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14212

File: build/cqs.tsv [hunk 13]
```diff
@@ -150,49 +148,50 @@ org.apache.parquet:parquet-encoding	1.9.0	compile
...
 org.codehaus.jackson:jackson-core-asl	1.9.3	compile
 org.codehaus.jackson:jackson-jaxrs	1.9.3	compile
 org.codehaus.jackson:jackson-mapper-asl	1.9.3	compile
-org.eclipse.emf:common	2.6.0	compile
-org.eclipse.emf:ecore	2.6.1	compile
-org.eclipse.xsd:xsd	2.6.0	compile
+org.eclipse.emf:org.eclipse.emf.common	2.12.0	compile
+org.eclipse.emf:org.eclipse.emf.ecore	2.12.0	compile
+org.eclipse.emf:org.eclipse.emf.ecore.xmi	2.12.0	compile
+org.eclipse.xsd:org.eclipse.xsd	2.12.0	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: eclipse projects shouldn't need a CQ - @jnh5y can you confirm these are ok?

File: build/cqs.tsv [hunk 14]
```diff
@@ -150,49 +148,50 @@ org.apache.parquet:parquet-encoding	1.9.0	compile
...
+org.geotools.ogc:net.opengis.ows	18-SNAPSHOT	compile
+org.geotools.ogc:net.opengis.wfs	18-SNAPSHOT	compile
+org.geotools.ogc:org.w3.xlink	18-SNAPSHOT	compile
+org.geotools.xsd:gt-xsd-core	18-SNAPSHOT	compile
+org.geotools.xsd:gt-xsd-fes	18-SNAPSHOT	compile
+org.geotools.xsd:gt-xsd-filter	18-SNAPSHOT	compile
+org.geotools.xsd:gt-xsd-gml2	18-SNAPSHOT	compile
+org.geotools.xsd:gt-xsd-gml3	18-SNAPSHOT	compile
+org.geotools.xsd:gt-xsd-ows	18-SNAPSHOT	compile
+org.geotools.xsd:gt-xsd-wfs	18-SNAPSHOT	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: will wait on geotools for the release i guess...

File: build/cqs.tsv [hunk 15]
```diff
@@ -210,8 +209,8 @@ org.mortbay.jetty:jetty-util	6.1.26	compile
 org.objenesis:objenesis	1.0	compile
 org.parboiled:parboiled-core	1.1.7	compile
 org.parboiled:parboiled-scala_2.11	1.1.7	compile
-org.scala-lang.modules:scala-parser-combinators_2.11	1.0.5	compile
-org.scala-lang.modules:scala-xml_2.11	1.0.5	compile
+org.scala-lang.modules:scala-parser-combinators_2.11	1.0.6	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14213

File: build/cqs.tsv [hunk 16]
```diff
@@ -210,8 +209,8 @@ org.mortbay.jetty:jetty-util	6.1.26	compile
...
+org.scala-lang.modules:scala-xml_2.11	1.0.6	compile
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14214

File: build/cqs.tsv [hunk 17]
```diff
@@ -231,40 +230,40 @@ com.datastax.cassandra:cassandra-driver-core	3.0.0	provided
...
-org.apache.hadoop:hadoop-mapreduce-client-core	2.5.2	provided
-org.apache.hadoop:hadoop-yarn-api	2.5.2	provided
-org.apache.hadoop:hadoop-yarn-common	2.5.2	provided
+org.apache.hadoop:hadoop-annotations	2.6.4	provided
+org.apache.hadoop:hadoop-client	2.6.4	provided
+org.apache.hadoop:hadoop-common	2.6.4	provided
+org.apache.hadoop:hadoop-distcp	2.6.4	provided
+org.apache.hadoop:hadoop-mapreduce-client-core	2.6.4	provided
+org.apache.hadoop:hadoop-yarn-api	2.6.4	provided
+org.apache.hadoop:hadoop-yarn-common	2.6.4	provided
```
Comment (elahrvivaz) [inline build/cqs.tsv]: done - should be covered under ~https://dev.eclipse.org/ipzilla/show_bug.cgi?id=8504~

File: build/cqs.tsv [hunk 18]
```diff
@@ -231,40 +230,40 @@ com.datastax.cassandra:cassandra-driver-core	3.0.0	provided
...
 org.apache.kafka:kafka-clients	0.10.0.1	provided
 org.apache.kafka:kafka-clients	0.9.0.1	provided
-org.apache.spark:spark-catalyst_2.11	2.0.0	provided
-org.apache.spark:spark-core_2.11	2.0.0	provided
-org.apache.spark:spark-sql_2.11	2.0.0	provided
-org.apache.spark:spark-yarn_2.11	2.0.0	provided
+org.apache.spark:spark-catalyst_2.11	2.2.0	provided
+org.apache.spark:spark-core_2.11	2.2.0	provided
+org.apache.spark:spark-sql_2.11	2.2.0	provided
+org.apache.spark:spark-yarn_2.11	2.2.0	provided
```
Comment (elahrvivaz) [inline build/cqs.tsv]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14215

**Comment by elahrvivaz** [comment]
we're going to cq geotools 18 when it's released in a couple weeks, and wait for our release until then.

**Comment by jahhulbert-ccri** [comment]
i think we can go ahead and merge this, no?

**Comment by jnh5y** [comment]
Nope.  @elahrvivaz wanted to wait on some CQs to go through.  Likely will go in next week or so.

**Comment by elahrvivaz** [comment]
a bunch came through already - just need to check they're all covered

File: build/cqs.tsv (hunk 6 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: this didn't pass the automated scanning...

File: build/cqs.tsv (hunk 8 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: also didn't pass the automated scanning

File: build/cqs.tsv (hunk 10 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: hasn't been pmc approved yet

File: build/cqs.tsv (hunk 15 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: waiting pmc

File: build/cqs.tsv (hunk 16 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: waiting pmc

File: build/cqs.tsv (hunk 18 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: didn't pass auto scan

**Comment by elahrvivaz** [comment]
looks like there are a few that are still waiting pmc, a few that failed IP scan and so have to be manually addressed by the IP team - not sure how long that will take

File: build/cqs.tsv (hunk 6 above)
Comment (jnh5y) [reply build/cqs.tsv]: Has been certified! (https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14205 has been updated)

File: build/cqs.tsv (hunk 16 above)
Comment (jnh5y) [reply build/cqs.tsv]: (Approved: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14214)

File: build/cqs.tsv (hunk 6 above)
Comment (elahrvivaz) [reply build/cqs.tsv]: nice, that was pretty fast

**Comment by elahrvivaz** [comment]
last four cqs here, for easier tracking:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14208
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14210
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14213
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14215

**Pull request merged**